WcaOnRails/lib/advance_to_next_round_requirement.rb

```diff
@@ -1,22 +1,35 @@
 # frozen_string_literal: true
 
 class AdvanceToNextRoundRequirement
+  include ActiveModel::Validations
+
+  ALLOWED_TYPES = %w(ranking percentile attemptValue).freeze
+
   attr_accessor :type, :ranking, :percentile, :attemptValue
+  validates :type, inclusion: { in: ALLOWED_TYPES }
+  validates :ranking, numericality: { only_integer: true }, allow_nil: true
+  validates :percentile, numericality: { only_integer: true }, allow_nil: true
+  validates :attemptValue, numericality: { only_integer: true }, allow_nil: true
+  validate do
+    errors.add(self.type.to_sym, "must be present") if self.value.nil?
+    errors.add(:ranking, "must not be present") if self.ranking.present? && self.type != "ranking"
+    errors.add(:percentile, "must not be present") if self.percentile.present? && self.type != "percentile"
+    errors.add(:attemptValue, "must not be present") if self.attemptValue.present? && self.type != "attemptValue"
```

Comment:
ALLOWED_TYPES.each do |allowed_type|
      if self.type != allowed_type
        errors.add(allowed_type, "must not be present") if self[allowed_type].present?
      end
    end